backend/src/Edna/DB/Connection.hs

```diff
@@ -63,3 +67,42 @@ createConnPool (ConnString connStr) maxConnsNum = liftIO $ ConnPool <$>
 -- | Destroys a @ConnPool@.
 destroyConnPool :: MonadIO m => ConnPool -> m ()
 destroyConnPool (ConnPool pool) = liftIO $ destroyAllResources pool
+
+-- | Provides methods for working with connections to Postgres database.
+data PostgresConn = PostgresConn
+  { _withConnection  :: forall (m :: Type -> Type) a. MonadUnliftIO m
+                     => (Connection -> m a) -> m a
+  , _withTransaction :: forall (m :: Type -> Type) a. MonadUnliftIO m
+                     => Connection -> m a  -> m a
+  }
+
+-- | Implementation of @PostgresConn@ which uses a @ConnPool@ to support
```

Comment:
Here I think you can use single quotes `'` instead of `@` because `PostgresConn` and `ConnPool` are actual identifiers in code. And below you should write `@withConnection@` instead of ``` `withConnection` ``` because these are not real identifiers (or write `'_withConnection'` for example).